bin/traceserver.py

```diff
@@ -116,8 +116,10 @@
 
 #    " -Xdebug -Xrunjdwp:transport=dt_socket,address=5005,server=y,suspend=n " + \
 #    " -XX:+UnlockCommercialFeatures -XX:+FlightRecorder -XX:FlightRecorderOptions=defaultrecording=true,dumponexit=true" + \
-java_cmd = '%(java)s $PHOENIX_OPTS ' + \
-    '-cp ' + hbase_config_path + os.pathsep + phoenix_utils.phoenix_traceserver_jar + os.pathsep + phoenix_utils.phoenix_client_jar + \
+java_cmd = '%(java)s  ' + \
```

Comment:
Seems like `$PHOENIX_OPTS` was dropped. Was this not being interpolated correctly? Is this environment variable propagated into the traceserver java process?

Reply:
https://issues.apache.org/jira/browse/PHOENIX-2659
 I think it's your comment:
"Josh Elser added a comment - 12/Feb/16 03:23
ok, PHOENIX_OPTS isn't working because subprocess isn't popping a shell like sqlline does. It will automatically get added to PQS's environment already. We can just remove that.
Steve is spot-on for the argument parsing. It's just busted."
So, I just delete it.

Reply:
All in all, the server successfully starts and works correctly.

Reply:
The issue is that subprocess doesn't handle shell variable expansion "in the command". Subprocess has API to provide an environment to the process being run.

`$PHOENIX_OPTS` and `$PHOENIX_TRACESERVER_OPTS` should be placed into the subprocess's environment, not just dropped.

Reply:
Do I need to add something like this?

phoenix_opts= os.getenv('PHOENIX_OPTS', '')
......
if hbase_env.has_key('PHOENIX_OPTS'):
    opts = hbase_env['PHOENIX_OPTS']
.....
java_cmd = '%(java)s  ' + phoenix_opts + \
....

Reply:
Actually, looks like queryserver.py doesn't include `$PHOENIX_OPTS` so let's just leave that to clients. `$PHOENIX_TRACESERVER_OPTS` is inlined, so you're good. Sorry for the confusion.

Reply:
Ok, thank you